writer.go

```diff
@@ -313,7 +343,7 @@ func (p *MediaPlaylist) AppendSegment(seg *MediaSegment) error {
 	if p.TargetDuration < seg.Duration {
 		p.TargetDuration = math.Ceil(seg.Duration)
 	}
-	p.buf.Reset()
+	p.dirty = true
```

Comment:
Does this need to be protected by the lock ?

Reply:
No, it's locked in the calling function.